app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -894,13 +901,25 @@ public void onStatusBarHeightFetched(int statusBarHeight) {
             }
         });
 
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && Settings.getInstance(getContext()).shouldUseBiometrics()) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
+                Settings.getInstance(getContext()).shouldUseBiometrics() &&
+                Biometrics.INSTANCE.hasFingerprintHardware(getContext())) {
             displayBiometricPromptIfNeeded();
         } else {
             getView().setAlpha(1);
         }
     }
 
+    // Handles the edge case of a user removing all enrolled prints while auth was enabled
+    private void checkBiometricStillValid() {
+        // Disable biometrics if the user is no longer eligible due to un-enrolling fingerprints:
+        if (!Biometrics.INSTANCE.hasFingerprintHardware(getContext())) {
```

Comment:
I would probably move this behavior to the Biometrics class and then call it in MainActivity